src/main/java/io/anserini/search/SearchCollection.java

```diff
@@ -85,85 +77,102 @@
-    if (searchArgs.searchtweets) {
-      analyzer = new TweetAnalyzer();
-    } else {
-      analyzer = searchArgs.keepstop ? new EnglishAnalyzer(CharArraySet.EMPTY_SET) : new EnglishAnalyzer();
+    TopicReader tr;
+    SortedMap<K, Map<String, String>> topics;
+    try {
+      tr = (TopicReader) Class.forName("io.anserini.search.query." + args.topicReader + "TopicReader")
+          .getConstructor(Path.class).newInstance(topicsFile);
+      topics = tr.read();
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Unable to load topic reader: " + args.topicReader);
     }
 
-    Query filter = null;
-    for (Map.Entry<K, Map<String, String>> entry : topics.entrySet()) {
-      K qID = entry.getKey();
-      String queryString = entry.getValue().get(searchArgs.topicfield);
-      Query query = AnalyzerUtils.buildBagOfWordsQuery(FIELD_BODY, analyzer, queryString);
-
-      if (searchArgs.searchtweets) {
-        long queryTweetTime = Long.parseLong(entry.getValue().get("time"));
-        // do not cosider the tweets with tweet ids that are beyond the queryTweetTime
-        // <querytweettime> tag contains the timestamp of the query in terms of the
-        // chronologically nearest tweet id within the corpus
-        filter = LongPoint.newRangeQuery(TweetGenerator.StatusField.ID_LONG.name, 0L, queryTweetTime);
-        BooleanQuery.Builder builder = new BooleanQuery.Builder();
-        builder.add(filter, BooleanClause.Occur.FILTER);
-        builder.add(query, BooleanClause.Occur.MUST);
-        query = builder.build();
-      }
+    final String runTag = "Anserini_" + args.topicfield + "_" + (args.keepstop ? "KeepStopwords_" : "")
+        + FIELD_BODY + "_" + (args.searchtweets ? "SearchTweets_" : "") + similarity.toString();
 
-      // Figure out how to break the scoring ties.
-      TopDocs rs;
-      if (searchArgs.arbitraryScoreTieBreak) {
-        rs = searcher.search(query, searchArgs.hits);
-      } else if (searchArgs.searchtweets) {
-        // TODO: we need to build the proper tie-breaking code path for tweets.
-        rs = searcher.search(query, searchArgs.hits);
-      } else {
-        rs = searcher.search(query, searchArgs.hits, BREAK_SCORE_TIES_BY_DOCID, true, true);
-      }
+    PrintWriter out = new PrintWriter(Files.newBufferedWriter(Paths.get(args.output), StandardCharsets.US_ASCII));
 
-      List<String> queryTokens = AnalyzerUtils.tokenize(analyzer, queryString);
-      if (searchArgs.searchtweets) { // This is ugly, but we have to reform the tweet query here for reranking
-        query = AnalyzerUtils.buildBagOfWordsQuery(FIELD_BODY, analyzer, queryString);
+    for (Map.Entry<K, Map<String, String>> entry : topics.entrySet()) {
+      K qid = entry.getKey();
+      String queryString = entry.getValue().get(args.topicfield);
+
+      ScoredDocuments docs;
+      if (args.searchtweets) {
+        docs = search(searcher, qid, queryString, Long.parseLong(entry.getValue().get("time")));
```

Comment:
It seems that you would like to separate searchTweets from search for other collection, right?
How about naming them differently?

Reply:
Here's what I'm thinking - I believe searchTweets is actually just search with a temporal filter. So  to generalize we should have two search methods - search with and without Filter query. This can just be overloaded methods.

Thoughts?